src/php/Printer/TypePrinter/StringPrinter.php

```diff
@@ -30,6 +30,16 @@ public function __construct(
     ) {
     }
 
+    public static function nonReadable(bool $withColor = false): self
```

Comment:
Having these new `nonReadable` and `readable` named-constructors, does it make sense to make `private` the constructor?